vignettes/satterthwaite.Rmd

```diff
@@ -88,14 +88,8 @@ $\hat\nu(\hat\theta)$.
 
 #### Jacobian calculation
 
-Currently, we evaluate the gradient of $\Phi(\theta)$ (which is created
-as an R function using `h_covbeta_fun()`) with regards to $\theta$
-evaluated at $\hat\theta$, i.e. the Jacobian $\mathcal{J}(\hat\theta)$,
-numerically using the `jacobian` function from the `numDeriv` package
-and organize it as a list (of length $k$ where $k$ is the dimension of
-the variance parameter vector $\theta$) of $c\times p$ matrices where
-$p$ is the dimension of $\beta$. This is implemented in function
-`h_jac_list()`.
+Currently, we evaluate the gradient of $\Phi(\theta)$ through function `h_jac_list()`.
+It uses automatic differiention provided in `TMB`.
```

Comment:
```suggestion
It uses automatic differentiation provided in `TMB`.
```
Thanks @clarkliming , that is great!
Few questions:
- can we maybe here describe more details on the C++ side computation of the Jacobian? like how you would explain someone the code in jacobian.cpp.
- do we then no longer need the `numDeriv` package in `mmrm`, i.e. can we remove it from dependencies?
- we mentioned in the issue that the same code is used in the KR calculations, could we avoid duplicate code?
- please add NEWS entry

Reply:
updated

Reply:
The KR calculations and Jacobian share a lot of duplicated code mainly in the input data parsing (obtaining the values from mmrm_data) so I think it is fine to keep two separated function.

The derivatives are similar but hard to harmonize (if we use some function to control the output, it will be a little messy)